build-tools/packages/build-infrastructure/README.md

```diff
@@ -0,0 +1,235 @@
+another larger workspace, contained within a single-package release group.
+
+## Features
+
+### Git repo capabilities
+
+A Fluid repo is often contained within a Git repository, and some functionality expects to be used within a Git
+repository. Features that need to execute Git operations can asynchronously retrieve the SimpleGit instance using the
+`IFluidRepo.getGitRepository` API. If the Fluid repo is not within a Git repo, then that call will throw a
+`NotInGitRepository` exception that callers should handle appropriately. If they don't, though, the exception makes it
+clear what has happened.
+
+> [!NOTE]
+>
+> This design addresses a major problem with build-tools v0, which was that code often made assumptions that it was
+> operating within a Git repo. That's often true, and some fetures can and should only work in that context, but the
+> implementation attempted to load the Git functionality blindly and would fail outright outside a Git context. With
+> `IFluidRepo`, the Git integration is more loosely coupled and the APIs make it clearer that it is not safe to assume
+> the presence of a Git repo.
+
+### Package selection and filtering APIs
+
+The IFluidRepo object provides access to workspaces, release groups, and their constiuent packages, but often one wants
+to operate on a subset of all packages in the repo. To support this, build-infrastructure provides a selection and
+filtering API. Packages can be selected based on criteria like workspace and release group, and the lists can be further
+filtered by scope or private/not private. Advanced filtering not covered by the built-in filters can be implemented
+using `Array.prototype.filter` on the results of package selection.
+
+### Loading old config formats
+
+The `repoPackages` configuration currently used by fluid-build will be loaded if the newer `repoLayout` config can't be
+found. This is for back-compat only and will not be maintained indefinitely. Users should convert to `repoLayout` when
+possible.
+
+## Configuration
+
+Configuration for the repo layout is stored in a config file at the root of the repo. This can either be part of the
+`fluidBuild.config.cjs` file in the `repoLayout` property, or in an independent config file named
+`repoLayout.config.cjs` (or mjs).
+
+### Example
+
+The following example configures three workspaces demonstrating the three archetypes - a workspace with multiple release
+groups, a workspace with a single release group that contains multiple packages, and a workspace with a single release
+group that contains a single package.
+
+```js
+repoLayout: {
+  workspaces: {
+    // This is the name of the workspace which is how it's referenced in the API.
+    "client": {
+      // This workspace is rooted at the root of the Git repo.
+      directory: ".",
+      releaseGroups: {
+        // This key is the name of the release group. It must be unique across all
+        // release groups.
```

Comment:
I think this is correct? The other version could be interpreted in the same way, but I think this makes it clearer if so.
```suggestion
        // This key is the name of the release group. It must be unique across all
        // workspaces.
```

Reply:
Hmmm, I'll find a way to reword this. The overall point is that these are keys in a map, so all release groups must have unique names (across the set of all release groups).